include/zephyr/net/net_pkt_filter.h

```diff
@@ -296,6 +323,60 @@ extern npf_test_fn_t npf_size_inbounds;
 		.test.fn = npf_size_inbounds, \
 	}
 
+/** @cond INTERNAL_HIDDEN */
+
+struct npf_test_ip {
+	struct npf_test test;
+	uint8_t addr_family;
+	void *ipaddr;
+	uint32_t ipaddr_num;
+};
+
+extern npf_test_fn_t npf_ip_src_addr_match;
+extern npf_test_fn_t npf_ip_src_addr_unmatch;
+
+/** @endcond */
+
+/**
+ * @brief Statically define a "ip address allowlist" packet filter condition
+ *
+ * This tests if the packet source ip address matches any of the ip
+ * addresses contained in the provided set.
+ *
+ * @param _name Name of the condition
+ * @param _ip_addr_array Array of <tt>struct in_addr</tt> or <tt>struct in6_addr</tt> items to test
+ *against
+ * @param _ip_addr_num number of IP addresses in the array
+ * @param _af Addresses family type (AF_INET / AF_INET6) in the array
+ */
+#define NPF_IP_SRC_ADDR_ALLOWLIST(_name, _ip_addr_array, _ip_addr_num, _af) \
```

Comment:
I am just wondering that shouldn't we have network interface (either index value or the pointer) here as a parameter? I am thinking about use case where there are multiple network interfaces and we want to block certain addresses either in one specific network interface or all of them.

Reply:
@jukkar I thought about something like you mentioned, but as a additional rule, e.g. 
```suggestion
#define NPF_IP_SRC_ADDR_ALLOWLIST_4_IFC(_name, _ip_addr_array, _ip_addr_num, _af) \
```
Additionally a user can will make by himself custom rule based on the structure related to this macro.
We can add it now or in future, what is your opinion?

Reply:
Oops, missed this discussion before merging. @MarGasiorek and @jukkar sorry, and feel free to continue the discussion and follow-up with a PR.

Reply:
No worries, additional features can be added in future PRs.